app/controllers/dashboards/base_controller.rb

```diff
@@ -58,5 +60,30 @@ def entered
       render layout: !request.xhr?
     end
 
+    def months
+      months = {}
+      active_report_class.ordered.select(:id, :parameters).index_by(&:parameters).each do | key, report |
+        report.set_date_range
+        start_date = report.range.start
+        months[report.id] = "#{Date::MONTHNAMES[start_date.month]} #{start_date.year}"
+      end
+      months
+    end
+
+    def selected_report_id
+      params[:choose_report][:month].to_i rescue id_of_most_recent_report
+    end
+
+    def id_of_most_recent_report
+      # assumes that newer reports have higher ids -- may not hold on dev
+      months.max_by {|k, v| k}[0] # id is key of the map
```

Comment:
purely stylistic, but if you aren't using one of the variables in a block, it's very helpful to use an `_`, so this could become: 
```months.max_by {|k, _| k}[0]```
Then, also, just to be more ruby-ish, it should be ```months.max_by {|k, _| k}&.first```  This will prevent raising any errors if max_by returns an empty array.